lib/model/content.dart

```diff
@@ -1451,6 +1451,60 @@ class _ZulipContentParser {
     return tableNode ?? UnimplementedBlockContentNode(htmlNode: tableElement);
   }
 
+  List<BlockContentNode> parseMathBlocks(dom.Element element) {
+    assert(_debugParserContext == _ParserContext.block);
+    assert(element.localName == 'p');
+    assert((() {
+      final first = element.nodes.first;
+      return first is dom.Element
+        && first.localName == 'span'
+        && first.className == 'katex-display';
+    })());
+    final debugHtmlNode = kDebugMode ? element : null;
+
+    final blocks = <BlockContentNode>[];
+
+    // The case with the `<br>\n` can happen when at the end of a quote;
+    // it seems like a glitch in the server's Markdown processing,
+    // so hopefully there just aren't any further such glitches.
+    bool hasTrailingBreakNewline = false;
+    if (element.nodes
+          case [..., dom.Element(localName: 'br'), dom.Text(text: '\n')]
+    ) {
+      hasTrailingBreakNewline = true;
+    }
+
+    final length = hasTrailingBreakNewline
+      ? element.nodes.length - 2
+      : element.nodes.length;
+    for (var i = 0; i < length; i++) {
+      final child = element.nodes[i];
+
+      // If there are multiple <span class="katex-display"> nodes in a <p>
+      // each node is interleaved by '\n\n'. Whitespaces are ignored in HTML
+      // on web but each node has `display: block`, which renders each node
+      // on a newline. So do the same here.
+      if (child case dom.Text(text: '\n\n')) continue;
```

Comment:
I can see how the first sentence is relevant. Regarding "do the same here", `continue` itself does not seem to address the part about rendering each node with `display: block`, as that is more relevant to `blocks.add(MathBlockNode(texSource: texSource, debugHtmlNode: debugHtmlNode));`.